spec/services/device_tracking/create_device_spec.rb

```diff
@@ -6,24 +6,18 @@
   let(:remote_ip) { '1.2.3.4' }
   let(:user_agent) { 'Chrome/58.0.3029.110 Safari/537.36' }
   let(:uuid) { 'abc123' }
-  let(:device) { Device.all.first }
 
   it 'creates a new device' do
-    subject.call(user.id, remote_ip, user_agent, uuid)
-
-    expect_device_attribute(:cookie_uuid, uuid)
-    expect_device_attribute(:user_agent, user_agent)
-    expect_device_attribute(:last_ip, remote_ip)
+    device = subject.call(user.id, remote_ip, user_agent, uuid)
+    expect(device.cookie_uuid).to eq uuid
+    expect(device.user_agent).to eq user_agent
+    expect(device.last_ip).to eq remote_ip
```

Comment:
just to check my understanding, these are not related to the compound PII fingerprint, right?

Reply:
sorry, yes, meant to mention that. After some local load testing of devices into my test database this started failing so I fixed this as well.